Gems/Atom/RPI/Code/Source/RPI.Reflect/Image/StreamingImageAsset.cpp

```diff
@@ -13,6 +13,95 @@
 
 namespace AZ
 {
+    namespace Internal
+    {
+        template <AZ::RHI::Format>
+        float RetrieveFloatValue(const AZ::u8* mem, size_t index)
+        {
+            AZ_Assert(false, "Unsupported pixel format");
+        }
+
+        template <AZ::RHI::Format>
+        AZ::u32 RetrieveUintValue(const AZ::u8* mem, size_t index)
+        {
+            AZ_Assert(false, "Unsupported pixel format");
+        }
+
+        template <>
+        float RetrieveFloatValue<AZ::RHI::Format::Unknown>([[maybe_unused]] const AZ::u8* mem, [[maybe_unused]] size_t index)
+        {
+            return 0.0f;
+        }
+
+        template <>
+        AZ::u32 RetrieveUintValue<AZ::RHI::Format::Unknown>([[maybe_unused]] const AZ::u8* mem, [[maybe_unused]] size_t index)
+        {
+            return 0;
+        }
+
+        template <>
+        AZ::u32 RetrieveUintValue<AZ::RHI::Format::R8_UNORM>(const AZ::u8* mem, size_t index)
+        {
+            return mem[index] / static_cast<AZ::u32>(std::numeric_limits<AZ::u8>::max());
```

Comment:
I don't think we want the division on any of these.  For one thing, 254/255 cast to an int is 0. ;)  But I think at the pixel fetching level, we just want to return the pixel value, and make as few assumptions about usage as possible.  Though this raises an interesting question on what to return.  I kinda feel like we should return 8-bit numbers for 8-bit formats, 16-bit numbers for 16-bit formats, etc.  I'm not sure what this will mean in terms of API usage though, I guess the caller will need different logic based on pixel value format.

Reply:
I actually lean towards doing the division in this case to be consistent with shaders. This is why there are several formats for what is under the hood the exact same data. For instance R8_UNORM, R8_UNORM_SRGB, R8_UINT, R8_SNORM, R8_SINT are all 8 bit single channel textures, but interpreted different ways.

Reply:
Not sure I understand?  0/255 - 254/255 are all producing a return value of 0, and 255/255 produces a return value of 1.  How does the division make this consistent with shaders?

Reply:
So just noticing the return value of u32 which doesn't make sense in this case. UNORM means the 8 bit values will be turned into a float in the range of 0.0 to 1.0. UNORM / SNORM / SRGB formats should only support float return values. In that situation 0 = 0.0 and 255 = 1.0.

Reply:
There may be a case for allowing the reading of an integer value directly, ignoring the UNORM / SNORM / SRGB format qualifiers, but I think we'd want to somehow be clear that you're getting a value in a different format than the underlying format intended. That would also bring up a question as to what to do if you try to get a u32 for an SINT type.